tests/integration/test_logcollector/test_macos/test_macos_file_status_when_no_macos.py

```diff
@@ -0,0 +1,131 @@
+
+    for daemon in required_test_daemons:
+        control_service('start', daemon=daemon)
+
+    yield
+
+    for daemon in required_test_daemons:
+        control_service('stop', daemon=daemon)
+
+
+@pytest.fixture(scope='module')
+def handle_files():
+    """Truncate local_internals file and create dummy file to be monitored by logcollector, then it is deleted."""
+    truncate_file(WAZUH_LOCAL_INTERNAL_OPTIONS)
+
+    with open(dummy_file, 'w') as f:
+        pass
+
+    yield
+
+    os.remove(dummy_file) if os.path.exists(dummy_file) else None
+
+
+def test_macos_file_status_when_no_macos(handle_files,
+                                         configure_local_internal_options_module,
+                                         get_configuration,
+                                         configure_environment,
+                                         file_monitoring,
+                                         restart_required_test_daemons):
+    """Checks that logcollector does not store and removes, if exists, previous "macos"-formatted localfile data in the
+    file_status.json
+
+    Given a file_status.json that contains a valid combination of "settings" and "timestamp" of "macos", when starting
+    an agent that has no "macos" localfile configured on its ossec.conf file, it should happen that, when
+    file_status.json is updated after a certain time, no "macos" status should remain stored on the status file.
+
+    Raises:
+        TimeoutError: If the callbacks, that checks the expected logs, are not satisfied in the expected time.
+    """
+
+    file_status_json = {}
+
+    log_monitor.start(timeout=LOG_COLLECTOR_GLOBAL_TIMEOUT,
+                      callback=callback_logcollector_started(),
+                      error_message="Logcollector did not start")
+
+    # Check if json_status contains 'macos' data and if not insert it
+    if os.path.exists(LOGCOLLECTOR_FILE_STATUS_PATH):
+        file_status_json = read_json(LOGCOLLECTOR_FILE_STATUS_PATH)
+        if 'macos' not in file_status_json:
+            file_status_json['macos'] = {}
+            file_status_json['macos']['timestamp'] = '2021-10-22 04:59:46.796446-0700'
+            file_status_json['macos']['settings'] = 'message CONTAINS "testing"'
+            write_json_file(LOGCOLLECTOR_FILE_STATUS_PATH, file_status_json)
+    else:
+        # If the file does not exist, then is created and then macos data is added
+        with open(LOGCOLLECTOR_FILE_STATUS_PATH, 'w') as f:
+            pass
+        file_status_json['macos'] = {}
+        file_status_json['macos']['timestamp'] = '2021-10-22 04:59:46.796446-0700'
+        file_status_json['macos']['settings'] = 'message CONTAINS "testing"'
+        write_json_file(LOGCOLLECTOR_FILE_STATUS_PATH, file_status_json)
+
+    # Waits for file_status.json to be created, with a timeout about the time needed to update the file
+    wait_file(LOGCOLLECTOR_FILE_STATUS_PATH, LOG_COLLECTOR_GLOBAL_TIMEOUT)
```

Comment:
Why `LOGCOLLECTOR_FILE_STATUS_PATH` ?  Maybe 30 seconds is too much to wait

Reply:
The 30 seconds is the **maximum** time to wait for the file to be created, it does not mean it will wait 30 seconds if the file was already created. As the file is deleted at the start, we need to wait for it to be created or just check if it already exists.